### PR TITLE
feat: governance or strategist approve / revoke access + sweep

### PR DESCRIPTION
**Pull request by btcookies**

### CODE REVIEW

**Review by sajanrajdev — APPROVED** [review]
Changes to SettV4h.sol:
- Changes the approve/revoke Contract Access permissions to be callable by the `strategist`.
- Adds a sweep function that transfers assets to the `governance` (can't transfer want token).

Testing:
Included an integration test parametrized to all SettV4s to be upgraded that:
- Test proper upgrading of the logic
- All of the Vaults/Strats functions are still callable and work properly post upgrade
- New functions and features work as expected

LGTM

**Comment by sajanrajdev** [comment]
Note: It is good practice to add a brief description of the intended changes to the PR as well as a screenshot of the tests passing, specially any new tests added.

File: contracts/SettV4h.sol [hunk 1]
```diff
@@ -317,6 +317,14 @@ contract SettV4h is ERC20Upgradeable, SettAccessControlDefended, PausableUpgrade
         _unpause();
     }
 
+    function sweep(IERC20Upgradeable _token) external {
+        _onlyGovernanceOrStrategist();
+        require(address(_token) != address(0), "NULL_ADDRESS");
```
Comment (shuklaayush) [inline contracts/SettV4h.sol]: We probably don't need this check since `safeTransfer` would revert

**Review by shuklaayush — APPROVED** [review]
LGTM

File: contracts/SettV4h.sol (hunk 1 above)
Comment (GalloDaSballo) [reply contracts/SettV4h.sol]: True

**Review by GalloDaSballo — APPROVED** [review]
LGTM, same with #3 

Strategist should still be a multi-sig and per our review practices code should be reviewed as if the contract block wasn't there

File: contracts/SettV4h.sol (hunk 1 above)
Comment (btcookies) [reply contracts/SettV4h.sol]: Updated

**Pull request merged**